server/routes/dsl_route.ts

```diff
@@ -29,15 +29,23 @@ export function registerDslRoute(router: IRouter) {
       const { query1, query2 } = request.body;
       const actionName =
         query1 && query2 ? METRIC_ACTION.COMPARISON_SEARCH : METRIC_ACTION.SINGLE_SEARCH;
-      let resBody: SearchResultsResponse = {};
+      const resBody: SearchResultsResponse = {};
 
       if (query1) {
-        const { index, size, ...rest } = query1;
-        const params: RequestParams.Search = {
-          index,
-          size,
-          body: rest,
-        };
+        const { index, pipeline, size, ...rest } = query1;
+        const params: RequestParams.Search =
+          pipeline !== ''
+            ? {
+                index,
+                size,
+                body: rest,
+                search_pipeline: pipeline,
```

Comment:
does this work? it seems no `search_pipeline` field in `search` https://github.com/opensearch-project/opensearch-js/blob/ac880e9c84d0a53d5d350e19b74199dc39140e3a/api/requestParams.d.ts#L1497C14-L1545